dali/python/nvidia/dali/plugin/tf.py

```diff
@@ -519,29 +576,36 @@ def __init__(self, pipeline, **kwargs):
         _insert_experimental_member(DALIDatasetWithInputs, "DALIDatasetWithInputs")
 
         class Input:
-            """Wrapper for input passed to DALIDataset. Allows to pass additional options.
+            """Wrapper for input passed to DALIDataset. Allows to pass additional options that
+            can override some of the ones provided to the External Source node in the
+            Python Pipeline object. Passing None indicates, that the value should be looked up
+            in the Pipeline defintion.
 
             Parameters
             ----------
             dataset : tf.data.Dataset
                 The dataset used as input
             layout : str, optional, default = None
-                Layout of given input. If None, the layout will be taken from the
-                Python Pipeline object. If it is not provided there, empty layout will be used.
+                Layout of given input. If None, the layout will be taken from the corresponding
+                External Source node in the Python Pipeline object. If it is not provided there,
+                empty layout will be used.
+            batch: bool, optional, default = False
+                Batch mode of given input. If None, the batch mode will be taken from the
+                corresponding External Source node in the Python Pipeline object.
+
+                If the ``batch = False``, the input dataset is considered sample input.
 
-                .. warning::
-                    Layout is not preserved by pipeline serialization and deserialization.
-                    If you are using deserialized pipeline and want to specify layout,
-                    you need to do it here.
+                If the ``batch = True``, the input dataset is expected to return batches.
             """
-            def __init__(self, dataset, *, layout=None):
+            def __init__(self, dataset, *, layout=None, batch=False):
```

Comment:
```suggestion
            def __init__(self, dataset, *, layout=None, batched=False):
```
?

Reply:
I want to keep it the same as External Source parameters.